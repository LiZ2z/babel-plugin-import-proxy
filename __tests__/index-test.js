const babel = require('babel-core');
const plugin = require('../index.js');


var example = `
import {Button, Table} from 'components';
import {Affix, Input} from 'antd';
import axios from 'axios';
`;


it('work without from key', () => {
    const { code } = babel.transform(example, {
        plugins: [
            [
                plugin,
                {
                    to: 'components/es/${name}',
                },
            ],
        ],
    });
    expect(code).toMatchSnapshot();
});


it('work without to key', () => {
    const { code } = babel.transform(example, {
        plugins: [
            [
                plugin,
                {
                    from: 'components',
                },
            ],
        ],
    });
    expect(code).toMatchSnapshot();
});

it('work with to key', () => {
    const { code } = babel.transform(example, {
        plugins: [
            [
                plugin,
                {
                    from: 'components',
                    to: 'components/es/${name}',
                },
            ],
        ],
    });
    expect(code).toMatchSnapshot();
});

it('work use CamelCase & without to key', () => {
    const { code } = babel.transform(example, {
        plugins: [
            [
                plugin,
                {
                    from: 'components',
                    useCamelCase: true,
                },
            ],
        ],
    });
    expect(code).toMatchSnapshot();
});

it('work with style key & without to key & use CamelCase', () => {
    const { code } = babel.transform(example, {
        plugins: [
            [
                plugin,
                {
                    from: 'components',
                    useCamelCase: true,
                    style: 'components/es/${name}/style.less',
                },
            ],
        ],
    });
    expect(code).toMatchSnapshot();
});

it('work with multiple plugins', () => {
    const { code } = babel.transform(example, {
        plugins: [
            [
                plugin,
                {
                    from: 'components',
                    useCamelCase: true,
                    style: 'components/es/${name}/style.less',
                },
            ],
            [
                plugin,
                {
                    from: 'antd',
                    to: 'antd/es/${name}',
                    style: 'antd/es/${name}/style.less',
                },
            ]
        ],
    });
    expect(code).toMatchSnapshot();
});