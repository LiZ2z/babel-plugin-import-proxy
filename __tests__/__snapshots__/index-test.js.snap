// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`work use CamelCase & without to key 1`] = `
"import Button from 'components/lib/Button';
import Table from 'components/lib/Table';

import { Affix, Input } from 'antd';
import axios from 'axios';"
`;

exports[`work with multiple plugins 1`] = `
"import Button from 'components/lib/Button';
import 'components/es/Button/style.less';
import Table from 'components/lib/Table';
import 'components/es/Table/style.less';
import Affix from 'antd/es/affix';
import 'antd/es/affix/style.less';
import Input from 'antd/es/input';
import 'antd/es/input/style.less';

import axios from 'axios';"
`;

exports[`work with style key & without to key & use CamelCase 1`] = `
"import Button from 'components/lib/Button';
import 'components/es/Button/style.less';
import Table from 'components/lib/Table';
import 'components/es/Table/style.less';

import { Affix, Input } from 'antd';
import axios from 'axios';"
`;

exports[`work with to key 1`] = `
"import Button from 'components/es/button';
import Table from 'components/es/table';

import { Affix, Input } from 'antd';
import axios from 'axios';"
`;

exports[`work without from key 1`] = `
"
import { Button, Table } from 'components';
import { Affix, Input } from 'antd';
import axios from 'axios';"
`;

exports[`work without to key 1`] = `
"import Button from 'components/lib/button';
import Table from 'components/lib/table';

import { Affix, Input } from 'antd';
import axios from 'axios';"
`;
